modules/network-observability-flows-format.adoc

```diff
@@ -0,0 +1,246 @@
+
+'''
+
+DstK8S_Namespace::
+
+• `Optional` *DstK8S_Namespace*: `string`
+
+Destination namespace
+
+'''
+
+SrcK8S_OwnerName::
+
+• `Optional` *SrcK8S_OwnerName*: `string`
+
+Source owner, such as Deployment, StatefulSet, etc.
+
+'''
+
+DstK8S_OwnerName::
+
+• `Optional` *DstK8S_OwnerName*: `string`
+
+Destination owner, such as Deployment, StatefulSet, etc.
+
+'''
+
+FlowDirection::
+
+• *FlowDirection*: <<Enumeration: FlowDirection,`FlowDirection`>>
+
+Flow direction from the node observation point
+
+
+== Fields
+
+
+SrcAddr::
+
+• *SrcAddr*: `string`
+
+Source IP address (ipv4 or ipv6)
+
+'''
+
+DstAddr::
+
+• *DstAddr*: `string`
+
+Destination IP address (ipv4 or ipv6)
+
+'''
+
+SrcMac::
+
+• *SrcMac*: `string`
+
+Source MAC address
+
+'''
+
+DstMac::
+
+• *DstMac*: `string`
+
+Destination MAC address
+
+'''
+
+SrcK8S_Name::
+
+• `Optional` *SrcK8S_Name*: `string`
+
+Name of the source matched Kubernetes object, such as Pod name, Service name, etc.
+
+'''
+
+DstK8S_Name::
+
+• `Optional` *DstK8S_Name*: `string`
+
+Name of the destination matched Kubernetes object, such as Pod name, Service name, etc.
+
+'''
+
+SrcK8S_Type::
+
+• `Optional` *SrcK8S_Type*: `string`
+
+Kind of the source matched Kubernetes object, such as Pod, Service, etc.
+
+'''
+
+DstK8S_Type::
+
+• `Optional` *DstK8S_Type*: `string`
+
+Kind of the destination matched Kubernetes object, such as Pod name, Service name, etc.
+
+'''
+
+SrcPort::
+
+• *SrcPort*: `number`
+
+Source port
+
+'''
+
+DstPort::
+
+• *DstPort*: `number`
+
+Destination port
+
+'''
+
+SrcK8S_OwnerType::
+
+• `Optional` *SrcK8S_OwnerType*: `string`
+
+Kind of the source Kubernetes owner, such as Deployment, StatefulSet, etc.
+
+'''
+
+DstK8S_OwnerType::
+
+• `Optional` *DstK8S_OwnerType*: `string`
+
+Kind of the destination Kubernetes owner, such as Deployment, StatefulSet, etc.
+
+'''
+
+SrcK8S_HostIP::
+
+• `Optional` *SrcK8S_HostIP*: `string`
+
+Source node IP
+
+'''
+
+DstK8S_HostIP::
+
+• `Optional` *DstK8S_HostIP*: `string`
+
+Destination node IP
+
+'''
+
+SrcK8S_HostName::
+
+• `Optional` *SrcK8S_HostName*: `string`
+
+Source node name
+
+'''
+
+DstK8S_HostName::
+
+• `Optional` *DstK8S_HostName*: `string`
+
+Destination node name
+
+'''
+
+Proto::
+
+• *Proto*: `number`
+
+L4 protocol
+
+'''
+
+Packets::
+
+• *Packets*: `number`
+
+Number of packets in this flow
+
+'''
+
+Bytes::
+
+• *Bytes*: `number`
+
+Number of bytes in this flow
+
+'''
+
+TimeFlowStartMs::
+
+• *TimeFlowStartMs*: `number`
+
+Start timestamp of this flow, in milliseconds
+
+'''
+
+TimeFlowEndMs::
+
+• *TimeFlowEndMs*: `number`
+
+End timestamp of this flow, in milliseconds
+
+'''
+
+TimeReceived::
+
+• *TimeReceived*: `number`
+
+Timestamp when this flow was received and processed by the flow collector, in seconds
+
+
+== Enumeration: FlowDirection
+
+
+Ingress::
+
+• *Ingress* = `"0"`
+
+Incoming traffic, from node observation point
+
+'''
+
+Egress::
+
+• *Egress* = `"1"`
+
+Outgoing traffic, from node observation point
```

Comment:
The document is twice as long because it displays the label or field and then repeats that in the bullet item.  This looks like a four-column table.  Something like:

<table><thead><tr><th>Label</th><th>Type</th><th>Optional?</th><th>Description</th></tr></thead><tbody><tr><td><strong>SrcK8S_Namespace</strong></td><td><code>string</code></td><td>&nbsp; &nbsp; &nbsp; &#10007;</td><td>Source namespace</td></tr><tr><td><strong>DstK8S_Namespace</strong></td><td><code>string</code></td><td>&nbsp; &nbsp; &nbsp; &#10007;</td><td>Destination namespace</td></tr></tbody></table>

Reply:
I see your point about repeating...We would need two tables if we did the above format. @jotak is it possible to generate two tables for Labels and Fields instead of having the list format?

Reply:
yeah, @stleerh , this doc is auto-generated upstream. Would need to investigate on formatting options. It uses [typedoc](https://typedoc.org/) under the hood with a markdown plugin: https://github.com/netobserv/network-observability-console-plugin/blob/main/web/package.json#L26 (and that's later translated into asciidoc).
So need to check if those tools support that, or alternatively if it can be done with some `sed`'ing ... we already do some here: https://github.com/netobserv/network-observability-operator/blob/main/hack/asciidoc-flows-gen.sh

Reply:
(to clarify my point, I think we can investigate to improve the formatting, but this shouldn't block this PR, right?)

Reply:
Agree, its non-blocking.

Reply:
Agree